### DISCUSSION TITLE
Background tasks

### DISCUSSION THREAD

**Discussion by RakeshPotnuru** [trpc/trpc#5990] [Q&A]
Hi, how to return response and do some background tasks. Like this

```js
app.get('/', (req, res) => {
  res.on('finish', () => {
    console.log('Response finished sending');
    // Perform cleanup tasks, background tasks, etc.
  });

  res.send('Hello World!');
});
```

_Repost from discord_

**Comment by KATT** [comment]
https://trpc.io/docs/server/middlewares#logging

**Comment by RakeshPotnuru** [comment]
Not middleware. Something I can do in an individual request. (sorry for confusing you with logging in comment)

do something -> return response -> then do something else after the request is completed (don't want the client to wait for this to complete)
